#! /usr/bin/env python

PACKAGE = "dummy_diag_publisher"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

Status = gen.enum(
    [
        gen.const("OK", int_t, 0, "OK"),
        gen.const("WARN", int_t, 1, "WARN"),
        gen.const("ERROR", int_t, 2, "ERROR"),
        gen.const("STALE", int_t, 3, "STALE"),
    ],
    "status of DiagnosticStatus.msg",
)

gen.add("status", int_t, 0, "status of DiagnosticStatus.msg", 0, 0, 3, edit_method=Status)
gen.add("is_active", bool_t, 0, "whether to send message", True)

exit(gen.generate(PACKAGE, "dummy_diag_publisher", "DummyDiagPublisher"))
